internal/resource/common/authenticationpolicytreenode/authentication_policy_tree_node_state.go

```diff
@@ -44,8 +44,8 @@ func childrenAttrTypes(depth int) types.ListType {
 	return childrenAttrTypesByDepth[depth]
 }
 
-func State(ctx context.Context, node *client.AuthenticationPolicyTreeNode) (types.Object, diag.Diagnostics) {
-	return recursiveState(ctx, node, 1, getRootNodeAttrTypes())
+func ToState(ctx context.Context, node *client.AuthenticationPolicyTreeNode, depth int) (types.Object, diag.Diagnostics) {
```

Comment:
Why was depth added as a parameter here? This method always expects to be called from the root node, which has a depth of 1.